tests/test_avroAlertGenerator.py

```diff
@@ -0,0 +1,456 @@
+            obshistid = obs.OpsimMetaData['obsHistID']
+            cat = TestAlertsTruthCat_avro(star_db, obs_metadata=obs)
+            cat.camera = lsst_camera()
+
+            for line in cat.iter_catalog():
+                if line[1] is None:
+                    continue
+
+                dmag = line[2]
+                mag = line[3]
+                if np.abs(dmag)>dmag_cutoff and mag<=self.obs_mag_cutoff[mag_name_to_int[obs.bandpass]]:
+                    alertId = (line[0]<<obshistid_bits) + obshistid
+                    assert alertId not in true_alert_dict
+                    true_alert_dict[alertId] = {}
+                    true_alert_dict[alertId]['chipName'] = line[1]
+                    true_alert_dict[alertId]['dmag'] = dmag
+                    true_alert_dict[alertId]['mag'] = mag
+                    true_alert_dict[alertId]['ra'] = np.degrees(line[4])
+                    true_alert_dict[alertId]['decl'] = np.degrees(line[5])
+                    true_alert_dict[alertId]['xPix'] = line[6]
+                    true_alert_dict[alertId]['yPix'] = line[7]
+
+        self.assertGreater(len(true_alert_dict), 10)
+
+        log_file_name = tempfile.mktemp(dir=self.alert_data_output_dir, suffix='log.txt')
+        alert_gen = AlertDataGenerator(testing=True)
+
+        alert_gen.subdivide_obs(self.obs_list, htmid_level=6)
+
+        for htmid in alert_gen.htmid_list:
+            alert_gen.alert_data_from_htmid(htmid, star_db,
+                                            photometry_class=TestAlertsVarCat_avro,
+                                            output_prefix='alert_test',
+                                            output_dir=self.alert_data_output_dir,
+                                            dmag_cutoff=dmag_cutoff,
+                                            log_file_name=log_file_name)
+
+        obshistid_to_htmid = {}
+        for htmid in alert_gen.htmid_list:
+            for obs in alert_gen.obs_from_htmid(htmid):
+                obshistid = obs.OpsimMetaData['obsHistID']
+                if obshistid not in obshistid_to_htmid:
+                    obshistid_to_htmid[obshistid] = []
+                obshistid_to_htmid[obshistid].append(htmid)
+
+        avro_gen = AvroAlertGenerator()
+        avro_gen.load_schema(os.path.join(getPackageDir('sims_catUtils'), 'tests', 'testData', 'avroSchema'))
+        sql_prefix_list = ['alert_test']
+        out_prefix = 'test_avro'
+        log_file_name = tempfile.mktemp(dir=self.avro_out_dir,
+                                        prefix='test_avro',
+                                        suffix='log.txt')
+        for obshistid in obshistid_list:
+            avro_gen.write_alerts(obshistid, self.alert_data_output_dir,
+                                  sql_prefix_list,
+                                  obshistid_to_htmid[obshistid],
+                                  self.avro_out_dir, out_prefix,
+                                  dmag_cutoff, lock=None,
```

Comment:
Since this `dmag_cutoff` can be different than the one used to create the catalog, do you think it would be useful to test using a different value?

Reply:
good idea.

I will add that to the test